src/Cart.ts

```diff
@@ -166,35 +166,88 @@ export default class Cart {
-					//		 check against `id` and `options`
-					items.push({
-						_id: String(p._id?? instance.items.length + 1),
-						id: String(p.id),
-						name: p.name,
-						price: Number(p.price),
-						quantity: Number(p.quantity || 1),
-						options: p.options || [],
-					});
+					// Check if item already exists then increment by quantity or 1,
+          //		 checking against `id` and `options`
+          let existingItem = instance.items.find(item => 
+            (item.id == p.id && 
+            (!item.options && !p.options) || 
+            ((item.options && p.options) && item.options!.length == p.options!.length &&
+            (item.options!.every(option => p.options!.indexOf(option) > -1))))
+          );
+
+          let itemQuantity : number = 0;
+          let itemPrice : number = 0;
+
+          // Parse and specify default quantity of 1
+          if(p.quantity && typeof p.quantity == 'string'){
+            itemQuantity = parseInt(p.quantity);
+          } else if(p.quantity && typeof p.quantity == 'number'){
+            itemQuantity = p.quantity;
+          } else {
+            itemQuantity = 1;
+          }
+
+          // Parse and round to 2 decimal places
+          if(p.price && typeof p.price == 'string'){
+            itemPrice = Math.round(parseFloat(p.price) * 100) / 100;
+          } else if(p.price && typeof p.price == 'number'){
+            itemPrice = Math.round(p.price * 100) / 100;
+          } else {
+            throw 'Price is undefined';
+          }
+
+          if(existingItem){
+            let itemOptions : CartUpdateOption;
+            
+            if(itemQuantity){
+              itemOptions = {
+                name: existingItem.name,
+                price: existingItem.price,
+                quantity: { value: itemQuantity, relative: true}
+              };
+            } else {
+              throw 'Quantity is undefined';
+            }
+            
+            existingItemOptions.push({id: existingItem.id, options: itemOptions});
+          } else {
+            items.push({
+              _id: String(p._id?? instance.items.length + 1),
+              id: String(p.id),
+              name: p.name,
+              price: itemPrice,
+              quantity: itemQuantity,
+              options: p.options || [],
+            });
+          }
 
 					if(p.conditions){
-						// instance.conditions.push.apply(null, p.conditions);
 						Array.prototype.push.apply(instance.conditions, p.conditions);
 					}
 				});
 
-				// instance.items.push.apply(null, items);
+        // Put new items into database
 				Array.prototype.push.apply(instance.items, items);
 				await storage.put(this._session, storage.serialise( this.compute(instance) ));
 
+        // Update existing items in database
+        // existingItemOptions.forEach(item => {
+        //   existingItems.push(await this.update(item.id, item.options));
+        // })
```

Comment:
Unable to update and store the updates in this array:

'await' expressions are only allowed within async functions and at the top levels of modules.

Could you suggest a way to call update for each of the items in this array? Or should another update function be written to accept arrays?